app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -18,6 +22,10 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
   val park: StateFlow<Park?>
     get() = _park
 
+  private val _parkLocation = MutableStateFlow<ParkLocation>(ParkLocation())
+  val parkLocation: StateFlow<ParkLocation>
+    get() = _parkLocation
+
```

Comment:
I'm not sure I understand the purpose of this variable. The `parkLocation` is already stored in the `_currentPark`, isn't it?

Reply:
This increases the separation between the fetching of the name and the creation of the park, as describe bellow. (Also if you test your implementation, you will notice that it does not work. I have at the moment no explanation why. Have you manage to have an implementation that works using your idea?)

Reply:
I understand your point of view, but I find it counterproductive to store the same variable twice. I've tried using the current park exclusively and managed to get a working version quickly and easily, we can talk about in person if you want!

Reply:
I think it is a good idea, it is better to solve this problem in person.

Reply:
After talking it over, we decided to leave the current implementation in place, as a change of implementation would require too much effort. I resolve this conversation and the ones related to.